### DISCUSSION TITLE
NVIDIA Audio Effects support?

### DISCUSSION THREAD

**Discussion by felipecrs** [wwmm/easyeffects#3208] [Ideas]
Thanks for this amazing app. It is really awesome and easy to use. I am opening this discussion because I did not see this being discussed elsewhere (and I wonder why).

It would be amazing if Easy Effects could provide effects based on NVIDIA Audio Effects:

https://docs.nvidia.com/deeplearning/maxine/audio-effects-sdk/index.html#getting-started-linux

They are generally much better than other noise suppression effects, for example, they can filter out barks.

This would make the most useful functionality of NVIDIA Broadcast for Windows available for Linux too.

**Comment by violetmage** [comment]
The [nvidia docs](https://docs.nvidia.com/deeplearning/maxine/audio-effects-sdk/index.html) that you reference explicitly say that the linux version is not intended for general deployment, instead only for server-side use where the environment is more controlled.

Additionally, even if it was robust enough for desktop use, it only works on a specific subset of NVIDIA GPUs. Limiting a feature to users with specific hardware would mean spending time working on something that not everyone can benefit from, which is not ideal for a project like easyeffects that is mostly hardware agnostic.

**Reply by felipecrs** [reply]
I see. Just for reference, OBS Studio integrates with NVIDIA Audio Effects SDK when it's installed (and a compatible NVIDIA card is found) (on Windows).

**Reply by wwmm** [reply]
>Additionally, even if it was robust enough for desktop use, it only works on a specific subset of NVIDIA GPUs. Limiting a feature to users with specific hardware would mean spending time working on something that not everyone can benefit from, which is not ideal for a project like easyeffects that is mostly hardware agnostic.

And on top of that I use AMD instead of Nvidia. Developing for a hardware blindly rarely ends well. I have to deal with something like this on another project and it is not great.

**Reply by felipecrs** [reply]
Oh, yes. Of course. I would not ask anyone to develop something they can't test.

I only expect us to agree whether this is a welcome feature or not.

**Reply by wwmm** [reply]
> Oh, yes. Of course. I would not ask anyone to develop something they can't test.
> 
> I only expect us to agree whether this is a welcome feature or not.

I think that having a feature tied to only one hardware provider is not the way to go. There aren't enough developers regularly contributing to EasyEffects to do the same kind of approach OBS and others do. If it were a solution based on OpenCL it would be more reasonable to spend effort on that.
